pytorch_lightning/utilities/distributed.py

```diff
@@ -345,3 +346,26 @@ def register_ddp_comm_hook(
 
 def tpu_distributed() -> bool:
     return _TPU_AVAILABLE and xm.xrt_world_size() > 1
+
+
+def init_ddp_connection(
+    cluster_environment: ClusterEnvironment,
+    torch_distributed_backend: str,
+    global_rank: Optional[int] = None,
+    world_size: Optional[int] = None,
+) -> None:
+    global_rank = global_rank if global_rank is not None else cluster_environment.global_rank()
+    world_size = world_size if world_size is not None else cluster_environment.world_size()
+    os.environ["MASTER_ADDR"] = cluster_environment.master_address()
+    os.environ["MASTER_PORT"] = str(cluster_environment.master_port())
```

Comment:
i forget why we set this. is this relied on elsewhere in lightning? is it specific to the lightning cluster environment?

Reply:
Yes, it's specific to the Cluster Environments.

Reply:
I believe this is required while creating the children with DDP ? @awaelchli  ?

Reply:
these are required for the `init_process_group` call below where we have `init_method='env://'`
the alternative is to do tcp initialization directly https://pytorch.org/docs/stable/distributed.html
Both ways are equivalent.

Unless something is broken here I would suggest not touching it ^^